forum/profiles/tests.py

```diff
@@ -618,15 +622,179 @@ def test_save_startup_wrong_startup_id(self):
         """Negative test to save startup to favourites when startup does not exist"""
         invalid_startup_id = StartupProfile.objects.order_by(
             '-id').first().id + 1 if StartupProfile.objects.exists() else 1
-        url = reverse('profiles:startups-save', kwargs={'pk': invalid_startup_id})
+        url = reverse(self.save_startup_url, kwargs={'pk': invalid_startup_id})
         self.client.credentials(HTTP_AUTHORIZATION=f'Bearer {self.token_user1}')
         response = self.client.post(url)
         self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
 
     def test_save_startup_second_time(self):
         """Negative test to save startup to favourites if startup is already followed"""
-        url = reverse('profiles:startups-save', kwargs={'pk': self.startup1.pk})
+        url = reverse(self.save_startup_url, kwargs={'pk': self.startup1.pk})
         self.client.credentials(HTTP_AUTHORIZATION=f'Bearer {self.token_user2}')
         self.client.post(url)
         response2 = self.client.post(url)
         self.assertEqual(response2.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_delete_favorite_startup_anonymous(self):
+        """Test that anonymous user has no access to the endpoint"""
+        url = reverse(self.unsave_startup_url, kwargs={'pk': self.startup1.pk})
+        response = self.client.post(url)
+        self.assertEqual(response.status_code, status.HTTP_401_UNAUTHORIZED)
+
+    def test_delete_favorite_startup(self):
+        """Test removing startup from saved favorites"""
+
+        # Saving startup first
+        save_url = reverse(self.save_startup_url, kwargs={'pk': self.startup1.pk})
+        self.client.credentials(HTTP_AUTHORIZATION=f'Bearer {self.token_user2}')
+        save_response = self.client.post(save_url)
+
+        # Unsaving startup
+        unsave_url = reverse(self.unsave_startup_url, kwargs={'pk': self.startup1.pk})
+        response = self.client.delete(unsave_url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
```

Comment:
Assert that the startup is removed from the investor's favorites by checking the database state.

Reply:
Done